pkg/steps/multi_stage/gen.go

```diff
@@ -41,7 +41,7 @@ func (s *multiStageTestStep) generateObservers(
 			Resources: observer.Resources,
 		})
 	}
-	pods, _, err := s.generatePods(adapted, nil, secretVolumes, secretVolumeMounts)
+	pods, _, err := s.generatePods(adapted, nil, secretVolumes, secretVolumeMounts, true)
```

Comment:
Since this is hardcoded as `true`, there is no need to pass it through arguments.

Reply:
Maybe it is better to create something like `struct PodOptions{ IsObserver bool}` as an optional argument, and you won't have to change the rest of the non-affected code.

Reply:
It sounds like a nice idea